update.sh

```diff
@@ -34,6 +44,84 @@ packages_found=
 packages_debs=
 packages_sha256=
 
+check_key() {
+    # param 1 = keyfile
+    # param 2 = key fingerprint
+
+    # check input parameters
+    if [ -z $1 ] || [ ! -f $1 ] ; then
+        echo "Parameter 1 of check_key() is not a file!"
+        return 1
+    fi
+
+    if [ -z $2 ] ; then
+        echo "Parameter 2 of check_key() is not a key fingerprint!"
+        return 1
+    fi
+
+    KEY_FILE="$1"
+    KEY_FINGERPRINT="$2"
+
+    echo -n "Checking key file '${KEY_FILE}'... "
+
+    # check that there is only 1 public key in the key file
+    if [ ! $(gpg --with-colons ${KEY_FILE} | grep ^pub: | wc -l) -eq 1 ] ; then
+        echo "FAILED!"
+        echo "There are zero or more than one keys in the ${KEY_FILE} key file!"
+        return 1
+    fi
+
+    # check that the key file's fingerprint is correct
+    if [ "$(gpg --with-colons ${KEY_FILE} | grep ^fpr: | awk -F: '{print $10}')" != "${KEY_FINGERPRINT}" ] ; then
+        echo "FAILED!"
+        echo "Bad GPG key fingerprint for ${KEY_FILE}!"
+        return 1
+    fi
+
+    echo "OK"
+    return 0
+}
+
+setup_archive_keys() {
+
+    mkdir -m 0700 -p gnupg
+
+    echo "Downloading ${RASPBIAN_ARCHIVE_KEY_FILE_NAME}."
+    curl -# -O ${RASPBIAN_ARCHIVE_KEY_URL}
+    if check_key "${RASPBIAN_ARCHIVE_KEY_FILE_NAME}" "${RASPBIAN_ARCHIVE_KEY_FINGERPRINT}" ; then
+        # GPG key checks out, thus import it into our own keyring
+        echo -n "Importing '${RASPBIAN_ARCHIVE_KEY_FILE_NAME}' into keyring... "
+        if gpg -q --homedir gnupg --import "${RASPBIAN_ARCHIVE_KEY_FILE_NAME}" ; then
+            echo "OK"
+        else
+            echo "FAILED!"
+            return 1
+        fi
+    else
+        return 1
+    fi
+
+    echo ""
+
+    echo "Downloading ${RASPBERRYPI_ARCHIVE_KEY_FILE_NAME}."
```

Comment:
Both keys are downloaded, checked, and imported in the same way. That process could be extracted into a function as well.